xray/backends/common.py

```diff
@@ -42,6 +46,28 @@ def is_trivial_index(var):
     return np.all(var.values == arange)
 
 
+def robust_getitem(array, key, catch=Exception, max_retries=6,
```

Comment:
Yep, I agree probably not worth adding a dependency.  If, in the future, we decided it was worth it the implementation is pretty easy/nice:

```
@retry(retry_on_exception=lambda e: isinstance(e, RuntimeError),
       wait_exponential_multiplier=500,
       stop_max_attempt_number=6)
def foo():
    return bar
```